tree/ntuple/v7/src/RFieldBase.cxx

```diff
@@ -620,7 +620,7 @@ std::size_t ROOT::Experimental::RFieldBase::ReadBulkImpl(const RBulkSpec &bulkSp
    std::size_t nRead = 0;
    for (std::size_t i = 0; i < bulkSpec.fCount; ++i) {
       // Value not needed
-      if (!bulkSpec.fMaskReq[i])
+      if (bulkSpec.fMaskReq && !bulkSpec.fMaskReq[i])
```

Comment:
I'd suggest caching `bulkSpec.fMaskReq` in a local variable. [Godbolt](https://godbolt.org/z/Y66vsqYs8) shows that it allows the compiler to optimize the loop better since it can know it's a loop invariant and basically compiles two different versions of the loop skipping the check for every iteration.

Reply:
Hm, I probably misread the assembly, but it stands to reason that the compiler will be able to create a more efficient version of the loop when having more information about the invariants, and I hardly consider an additional local variable as code complexity.

Reply:
We might be able to help the compiler a bit by putting both `fMaskReq` and `fMaskAvail` in local variables, but before I see performance data I wouldn't do it.

Reply:
Fine for me